manager/controllers/app/plotter_controller_test.go

```diff
@@ -0,0 +1,185 @@
+			deployAndCheck(utils.GetSystemNamespace(), true)
+		})
+	})
+})
+
+// Create and then Delete the plotter and check that the result is what was expected.
+// Deploy the plotter and check that the result is what was expected.
+func deployDeleteAndCheck(namespace string, shouldSucceed bool) {
+	const timeout = time.Second * 30
+	const interval = time.Millisecond * 100
+
+	// Create the plotter yaml from the hard coded testdata file
+	plotter := &app.Plotter{}
+	Expect(readObjectFromFile("../../testdata/plotter.yaml", plotter)).ToNot(HaveOccurred())
+
+	// Set the namespace to the system namespace so that the plotter will be successfully created
+	plotter.SetNamespace(utils.GetSystemNamespace())
+
+	// Create the Plotter.  Need to wait in case it's still in the process of being deleted from a previous test.
+	By("Expecting the plotter to be created.")
+	Eventually(func() error {
+		return k8sClient.Create(context.Background(), plotter)
+	}, timeout, interval).Should(Succeed())
+
+	// Don't forget to clean up after tests finish
+	plotterKey := client.ObjectKeyFromObject(plotter)
+	defer func() {
+		plotter := &app.Plotter{ObjectMeta: metav1.ObjectMeta{Namespace: plotterKey.Namespace, Name: plotterKey.Name}}
+		_ = k8sClient.Get(context.Background(), plotterKey, plotter)
+		_ = k8sClient.Delete(context.Background(), plotter)
+	}()
+
+	By("Expecting to find the plotter we created")
+	Eventually(func() error {
+		return k8sClient.Get(context.Background(), plotterKey, plotter)
+	}, timeout, interval).Should(Succeed())
+
+	By("Expecting Plotter to eventually be ready")
+	Eventually(func() bool {
+		Expect(k8sClient.Get(context.Background(), plotterKey, plotter)).To(Succeed())
+		return plotter.Status.ObservedState.Ready
+	}, timeout, interval).Should(BeTrue(), "Plotter is not ready after timeout!")
+
+	// Now create a request to delete the same plotter
+	// Set the namespace to the namespace being used for this test
+	delPlotter := &app.Plotter{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: plotterKey.Name}}
+
+	// Delete the Plotter
+	if shouldSucceed {
+		Expect(k8sClient.Delete(context.Background(), delPlotter)).Should(Succeed())
+	} else {
+		Expect(k8sClient.Delete(context.Background(), delPlotter)).ShouldNot(Succeed())
+
+		// Cleanup - from correct namespace
+		Expect(k8sClient.Delete(context.Background(), plotter)).Should(Succeed())
+	}
+}
+
+var _ = Describe("Plotter Controller Legal Delete Event", func() {
+	Context("Plotter", func() {
+		BeforeEach(func() {
+			// Add any setup steps that needs to be executed before each test
+		})
+
+		AfterEach(func() {
+			// Add any teardown steps that needs to be executed after each test
+		})
+
+		// Plotter is successfully deleted when deletion is done from m4d-system only
+		It("Test Plotter Deletion from Correct Namespace", func() {
+			deployDeleteAndCheck(utils.GetSystemNamespace(), true)
+		})
+	})
+})
+
+var _ = Describe("Plotter Controller Illegal Delete Event", func() {
```

Comment:
I think what we should check in the delete test is the following:
    create a Plotter in m4d-system
    create a similar Plotter in default
    delete Plotter in default
    check that a blueprint was not deleted
    delete Plotter in m4d-system
    check that a blueprint was deleted

Reply:
@shlomitk1 It's not possible to do the test you describe because there is a filter that prevents the reconcile loop from being called if the plotter is created from outside of m4d-system.  Thus, no blueprint will be created when Plotter is created in default.

Reply:
> 
> 
> @shlomitk1 It's not possible to do the test you describe because there is a filter that prevents the reconcile loop from being called if the plotter is created from outside of m4d-system. Thus, no blueprint will be created when Plotter is created in default.

This is exactly the intent of the test. Deleting a plotter in default is not supposed to delete the blueprint that was created by the "legal" plotter.

Reply:
@shlomitk1 I can't delete a plotter in default, because there is no way to successfully create one in default.

Reply:
You can create a plotter in default. A reconcile won't be called, but there is no relation between two. You can or can't create resources in different namespaces based on the rbac permissions only.